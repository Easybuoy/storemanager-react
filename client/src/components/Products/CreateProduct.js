import React, { Component } from 'react'
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import PropTypes from 'prop-types';

import { createProduct } from '../../actions/productActions';
import Loading from '../Common/Loading';

export class CreateProduct extends Component {
    constructor() {
        super();
        this.state = {
            productname: '',
            productdescription: '',
            productprice: '',
            productquantity: '',
            productimage: '',
            categoryoption: ''
        }
    }

    onChange = (event) => {
        this.setState({ [event.target.name]: event.target.value });
    }
 
    onImageChange = (event) => {
        this.setState({ productimage: event.target.files[0] });
    }

    onSubmit = (e) => {
        e.preventDefault();
        const { productname, productdescription, productprice, productquantity, productimage } = this.state;
        const productData = {
            name: productname,
            description: productdescription,
            price: productprice,
            quantity: productquantity,
            productimage
        }
        this.props.createProduct(productData);
    }

    render() {
        const { errors } = this.props;
        const { loading, isProductCreated, } = this.props.product;
        
        if (loading) {
            return (
                <div>
                <Loading />
                </div>
            )
        }
        if (isProductCreated) {
            toast.success('Product created successfully');
        }

        if (Object.keys(errors).length > 0) {
            if (errors.name) {
                toast.error(errors.name);
            }

            if (errors.description) {
                toast.error(errors.description);
            }

            if (errors.price) {
                toast.error(errors.price);
            }

            if (errors.quantity) {
                toast.error(errors.quantity);
            }

            if (errors.message) {
                toast.error(errors.message);
            }
        }

        

        return ( 
            <div id="main" className="topmargin">

                <div className="container">
                    <div id="create_sales_assistant">
                        <h1 className="text-center">Create New Product</h1>
                        <form onSubmit={this.onSubmit}>
                            <h2>Product Name</h2>
                            <input type="text" id="productname" name="productname" placeholder="Enter Name" onChange={this.onChange} required/>
                            <h2>Product Summary</h2>
                            <input type="text" id="productdescription" name="productdescription" placeholder="Enter Description" onChange={this.onChange} required/>
                            <h2>Product Price</h2>
                            <input type="number" id="productprice" name="productprice" placeholder="Enter Price" onChange={this.onChange} required/>
                            <h2>Product Quantity</h2>
                            <input type="number" id="productquantity" name="productquantity" placeholder="Enter Quantity" onChange={this.onChange} required/>
                            <img src="" id="showimage" className="text-center" style={{ display: 'none', height: '100px', width: '100px' }} />
                            <h2>Product Image</h2>
                            <input type="file" id="productimage" name="upload" onChange={this.onImageChange} />

                            <h2>Category</h2>
                            <select id="categoryoption">
                                <option value="">Select Category</option>
                            </select> <br /> <br />
                            <input type="submit" name="" className="button_1" id="createproductsubmit" value="Create Product"  />

                        </form>
                    </div>

                </div>
            </div>
        )
    }
}

CreateProduct.propTypes = {
    createProduct: PropTypes.func.isRequired,
    product: PropTypes.object.isRequired,
    errors: PropTypes.object.isRequired
  }

const mapStateToProps = state => ({
    product: state.products,
    errors: state.errors
});

export default connect(mapStateToProps, { createProduct })(CreateProduct);