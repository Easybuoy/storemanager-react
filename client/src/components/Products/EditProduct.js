import React, { Component } from 'react'
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import PropTypes from 'prop-types';

import { getProductById, editProduct } from '../../actions/productActions';
import Loading from '../Common/Loading';

export class EditProduct extends Component {
    componentDidMount = () => {
       this.props.getProductById(this.props.match.params.id);
      
    }

    onSubmit = (e) => { 
        e.preventDefault();
        const productname = this.refs.productname.value;
        const productdescription = this.refs.productdescription.value;
        const productquantity = this.refs.productquantity.value;
        const productprice = this.refs.productprice.value
        const productimage = this.refs.productimage.files[0];
    
        const productData = {
            name: productname,
            description: productdescription,
            price: productprice,
            quantity: productquantity,
            productimage
        } 
        this.props.editProduct(this.props.match.params.id, productData);
    }

    render() {
        const { errors } = this.props;
        const { isProductEdited, product } = this.props.product;
        
        if (isProductEdited) {
            toast.success('Product edited successfully');
            this.props.history.push('/dashboard');
        }

        if (Object.keys(errors).length > 0) {
            if (errors.message) {
                toast.error(errors.message);
            }
        }

            if (product) {
                return (
                    <div id="main" className="topmargin">

                <div className="container">
                    <div id="create_sales_assistant">
                        <h1 className="text-center">Edit Product</h1>
                        <form onSubmit={this.onSubmit}>
                            <h2>Product Name</h2>
                            <input type="text" ref="productname" id="productname" defaultValue={product.name} name="productname" placeholder="Enter Name" onChange={this.onChange} required/>
                            <h2>Product Summary</h2>
                            <input type="text" ref="productdescription" id="productdescription" defaultValue={product.description} name="productdescription" placeholder="Enter Description" onChange={this.onChange} required/>
                            <h2>Product Price</h2>
                            <input type="number" ref="productprice" id="productprice" name="productprice" defaultValue={product.price} placeholder="Enter Price" onChange={this.onChange} required/>
                            <h2>Product Quantity</h2>
                            <input type="number" ref="productquantity" id="productquantity" defaultValue={product.quantity} name="productquantity" placeholder="Enter Quantity" onChange={this.onChange} required/>
                            <img src="" id="showimage" className="text-center" style={{ display: 'none', height: '100px', width: '100px' }} />
                            <h2>Product Image</h2>
                            <input type="file" ref="productimage" id="productimage" name="upload" onChange={this.onImageChange} />

                            <h2>Category</h2>
                            <select id="categoryoption">
                                <option value="">Select Category</option>
                            </select> <br /> <br />
                            <input type="submit" name="" className="button_1" id="createproductsubmit" value="Edit Product"  />

                        </form>
                    </div>

                </div>
            </div>
                )
            }
        
        

        return ( 
            <div id="main" className="topmargin">
                <Loading />
            </div>
        )
    }
}

EditProduct.propTypes = {
    getProductById: PropTypes.func.isRequired,
    editProduct: PropTypes.func.isRequired,
    product: PropTypes.object.isRequired,
    errors: PropTypes.object.isRequired
  }

const mapStateToProps = state => ({
    product: state.products,
    errors: state.errors
});

export default connect(mapStateToProps, { getProductById, editProduct })(EditProduct);