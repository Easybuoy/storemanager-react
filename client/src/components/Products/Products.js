import React, { Component } from 'react'
import { connect } from 'react-redux';
import PropTypes from 'prop-types';


import Loading from '../Common/Loading';;

import { getProducts, deleteProduct } from '../../actions/productActions';
import { toast } from 'react-toastify';

class Products extends Component {

    componentDidMount(){
        this.showCartCount();
        this.props.getProducts();
    }

    deleteProduct(id) {
      if (confirm('Are you sure you want to delete this product?')){
        this.props.deleteProduct(id);
      };
      
    }

    addToCart(productId, productImage, productPrice) {
      const products = {
      productId,
      productImage,
      productPrice
    };

    const totalcartitems = JSON.parse(localStorage.getItem('products'));
    // If nothing is in cart add new item
    if (totalcartitems === null) {
      const data = [];
  
      data.push(products);
      localStorage.setItem('products', JSON.stringify(data));
      return this.showCartCount();
    }
    // Else push to what is existing in cart.
    totalcartitems.push(products);
    localStorage.setItem('products', JSON.stringify(totalcartitems));      
      this.showCartCount();
    }

    showCartCount() {
      let length = 0;
      const totalcartitem = JSON.parse(localStorage.getItem('products'));
      if (totalcartitem) {
      length = totalcartitem.length;
      }
      document.getElementById('shoppingcartlabel').innerHTML = length;
    }

  render() {
    const { products, loading, productDeleted } = this.props.products;
    
    if (Object.keys(this.props.errors).length > 0) {

      toast.error(this.props.errors.message)
      return (
        <div>
          <h1 className="text-center">
            {this.props.errors.message}
          </h1>
        </div>
      );
    }

    if (loading) { 
      return (
        <div>
        <Loading />
        </div>
      )
    }
   
    if (productDeleted) {
      toast.success('Product Deleted Successfully');
    }

    if (products) {
      // If type == 1, its an admin
      if (this.props.auth.user.type === 1) {
        return (
          <div className="container">
          <section id="categoryfiltersection">
            <div className="container">
                    <div className="categoryfiltercontent">
                        <select>
                            <option value="Phones & Tablets">Phones & Tablets</option>
                            <option value="Computing">Computing</option>
                            <option value="Gaming">Gaming</option>
                        </select>
                        <button className="button_1">Filter</button>
                       
                    </div>
            </div>
            </section>
            <div className="cardgroup">
            {
              products.map((product, key) => {
                const productid = product.id;
                const productName = product.name;
                const productDescription = `${product.description.substring(0, 70)}...`;
                const productPrice = product.price;
                const productQuantity = product.quantity;
                const productImage = `https://store--manager.herokuapp.com/${product.product_image}`;
                return (
                <div key={key} className="card">
                  <a href="view_product_details.html"><img src={productImage} className="cardimg" /></a>
          
                  <div className="text-center cardbody" >
                    <h3 id="productname">{productName}</h3>
                    <p>{productDescription} </p>
                    <p>Quantity: {productQuantity}</p>
                    <p  id="productamount">Price: {`$${productPrice}`}</p>
                    <button className="button_1"><a href="admin_edit_product.html?id=${product.id}">EDIT</a></button>
                    <button className="button_2" onClick={() => {this.deleteProduct(productid)}}>DELETE</button>
                </div>
              </div>
                );
              })
            }
              
            </div>
          </div>
        )
      }

      return (
        <div className="container">
        <section id="categoryfiltersection">
            <div className="container">
                    <div className="categoryfiltercontent">
                        <select>
                            <option value="Phones & Tablets">Phones & Tablets</option>
                            <option value="Computing">Computing</option>
                            <option value="Gaming">Gaming</option>
                        </select>
                        <button className="button_1">Filter</button>
                       
                    </div>
            </div>
            </section>
          <div className="cardgroup">
          {
            products.map((product, key) => {
              const productId = product.id;
              const productName = product.name;
              const productDescription = `${product.description.substring(0, 70)}...`;
              const productPrice = product.price;
              const productQuantity = product.quantity;
              const productImage = `https://store--manager.herokuapp.com/${product.product_image}`
              return (
              <div key={key} className="card">
                <a href="view_product_details.html"><img src={productImage} alt={productName} className="cardimg" /></a>
        
                <div className="text-center cardbody" >
                  <h3 id="productname">{productName}</h3>
                  <p>{productDescription} </p>
                  <p>Quantity: {productQuantity}</p>
                  <p  id="productamount">Price: {`$${productPrice}`}</p>
                  <button className="button_1" onClick={() => {this.addToCart(productId, productImage, productPrice)}}>Add To Cart</button>
              </div>
            </div>
              );
            })
          }
            
          </div>
        </div>
      )
    }

    return (
      <div>
        
      </div>
    )
  }
}

Products.propTypes = {
  getProducts: PropTypes.func.isRequired,
  deleteProduct: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  products: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired
}

const mapStateToProps = state => ({
    auth: state.auth,
    products: state.products,
    errors: state.errors
});

export default connect(mapStateToProps, { getProducts, deleteProduct })(Products);