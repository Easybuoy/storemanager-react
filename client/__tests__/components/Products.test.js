import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import Products from '../../src/components/Products/Products';

describe('<Products />', () => {
    const props = {
        getProducts: jest.fn(),
        deleteProduct: jest.fn(),
        auth: jest.mock(),
        products: jest.mock(),
        errors: jest.mock(),
      };

    it('renders the Products component correctly', () => {
        const wrapper = shallow(<Products {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    // it('should call the mock getProducts function', () => {
    //     const wrapper = shallow(<Products {...props} />)
 
    //     //  wrapper.find('form').simulate(
    //     //    'submit', 
    //     //    {preventDefault() {}}
    //     //  )
    //      expect(props.getProducts).toBeCalled();
    //      expect(props.getProducts).toHaveReturned();
    //     });

    // it('should call the mock addToCart function', () => {
    //     const wrapper = shallow(<Products {...props} />)

    //     const event = {target: {name: "pollName", value: "spam"}};

    //      sinon.spy(wrapper.instance(), 'getProducts');
    //      wrapper.instance().getProducts();
    //      expect(wrapper.instance().getProducts.calledOnce)
    //        .toEqual(true);
    //     //    expect(wrapper.instance().onChange.calledWith(event));

    //     });

    it('should render productRecord if sales array is not empty', () => {
        props.products.products = [{id: 1, name: 'iphone'}];
        const wrapper = shallow(<Products {...props} />)

        expect(wrapper).toMatchSnapshot();
        });
});