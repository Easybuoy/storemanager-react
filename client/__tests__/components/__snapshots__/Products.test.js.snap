// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Products /> renders the Products component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Products)
    auth={
      Object {
        "addMatchers": [Function],
        "advanceTimersByTime": [Function],
        "autoMockOff": [Function],
        "autoMockOn": [Function],
        "clearAllMocks": [Function],
        "clearAllTimers": [Function],
        "deepUnmock": [Function],
        "disableAutomock": [Function],
        "doMock": [Function],
        "dontMock": [Function],
        "enableAutomock": [Function],
        "fn": [Function],
        "genMockFromModule": [Function],
        "isMockFunction": [Function],
        "mock": [Function],
        "requireActual": [Function],
        "requireMock": [Function],
        "resetAllMocks": [Function],
        "resetModuleRegistry": [Function],
        "resetModules": [Function],
        "restoreAllMocks": [Function],
        "retryTimes": [Function],
        "runAllImmediates": [Function],
        "runAllTicks": [Function],
        "runAllTimers": [Function],
        "runOnlyPendingTimers": [Function],
        "runTimersToTime": [Function],
        "setMock": [Function],
        "setTimeout": [Function],
        "spyOn": [Function],
        "unmock": [Function],
        "useFakeTimers": [Function],
        "useRealTimers": [Function],
      }
    }
    deleteProduct={[MockFunction]}
    errors={
      Object {
        "addMatchers": [Function],
        "advanceTimersByTime": [Function],
        "autoMockOff": [Function],
        "autoMockOn": [Function],
        "clearAllMocks": [Function],
        "clearAllTimers": [Function],
        "deepUnmock": [Function],
        "disableAutomock": [Function],
        "doMock": [Function],
        "dontMock": [Function],
        "enableAutomock": [Function],
        "fn": [Function],
        "genMockFromModule": [Function],
        "isMockFunction": [Function],
        "mock": [Function],
        "requireActual": [Function],
        "requireMock": [Function],
        "resetAllMocks": [Function],
        "resetModuleRegistry": [Function],
        "resetModules": [Function],
        "restoreAllMocks": [Function],
        "retryTimes": [Function],
        "runAllImmediates": [Function],
        "runAllTicks": [Function],
        "runAllTimers": [Function],
        "runOnlyPendingTimers": [Function],
        "runTimersToTime": [Function],
        "setMock": [Function],
        "setTimeout": [Function],
        "spyOn": [Function],
        "unmock": [Function],
        "useFakeTimers": [Function],
        "useRealTimers": [Function],
      }
    }
    getProducts={[MockFunction]}
    products={
      Object {
        "addMatchers": [Function],
        "advanceTimersByTime": [Function],
        "autoMockOff": [Function],
        "autoMockOn": [Function],
        "clearAllMocks": [Function],
        "clearAllTimers": [Function],
        "deepUnmock": [Function],
        "disableAutomock": [Function],
        "doMock": [Function],
        "dontMock": [Function],
        "enableAutomock": [Function],
        "fn": [Function],
        "genMockFromModule": [Function],
        "isMockFunction": [Function],
        "mock": [Function],
        "requireActual": [Function],
        "requireMock": [Function],
        "resetAllMocks": [Function],
        "resetModuleRegistry": [Function],
        "resetModules": [Function],
        "restoreAllMocks": [Function],
        "retryTimes": [Function],
        "runAllImmediates": [Function],
        "runAllTicks": [Function],
        "runAllTimers": [Function],
        "runOnlyPendingTimers": [Function],
        "runTimersToTime": [Function],
        "setMock": [Function],
        "setTimeout": [Function],
        "spyOn": [Function],
        "unmock": [Function],
        "useFakeTimers": [Function],
        "useRealTimers": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`<Products /> should render productRecord if sales array is not empty 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Products)
    auth={
      Object {
        "addMatchers": [Function],
        "advanceTimersByTime": [Function],
        "autoMockOff": [Function],
        "autoMockOn": [Function],
        "clearAllMocks": [Function],
        "clearAllTimers": [Function],
        "deepUnmock": [Function],
        "disableAutomock": [Function],
        "doMock": [Function],
        "dontMock": [Function],
        "enableAutomock": [Function],
        "fn": [Function],
        "genMockFromModule": [Function],
        "isMockFunction": [Function],
        "mock": [Function],
        "products": Array [
          Object {
            "id": 1,
            "name": "iphone",
          },
        ],
        "requireActual": [Function],
        "requireMock": [Function],
        "resetAllMocks": [Function],
        "resetModuleRegistry": [Function],
        "resetModules": [Function],
        "restoreAllMocks": [Function],
        "retryTimes": [Function],
        "runAllImmediates": [Function],
        "runAllTicks": [Function],
        "runAllTimers": [Function],
        "runOnlyPendingTimers": [Function],
        "runTimersToTime": [Function],
        "setMock": [Function],
        "setTimeout": [Function],
        "spyOn": [Function],
        "unmock": [Function],
        "useFakeTimers": [Function],
        "useRealTimers": [Function],
      }
    }
    deleteProduct={[MockFunction]}
    errors={
      Object {
        "addMatchers": [Function],
        "advanceTimersByTime": [Function],
        "autoMockOff": [Function],
        "autoMockOn": [Function],
        "clearAllMocks": [Function],
        "clearAllTimers": [Function],
        "deepUnmock": [Function],
        "disableAutomock": [Function],
        "doMock": [Function],
        "dontMock": [Function],
        "enableAutomock": [Function],
        "fn": [Function],
        "genMockFromModule": [Function],
        "isMockFunction": [Function],
        "mock": [Function],
        "products": Array [
          Object {
            "id": 1,
            "name": "iphone",
          },
        ],
        "requireActual": [Function],
        "requireMock": [Function],
        "resetAllMocks": [Function],
        "resetModuleRegistry": [Function],
        "resetModules": [Function],
        "restoreAllMocks": [Function],
        "retryTimes": [Function],
        "runAllImmediates": [Function],
        "runAllTicks": [Function],
        "runAllTimers": [Function],
        "runOnlyPendingTimers": [Function],
        "runTimersToTime": [Function],
        "setMock": [Function],
        "setTimeout": [Function],
        "spyOn": [Function],
        "unmock": [Function],
        "useFakeTimers": [Function],
        "useRealTimers": [Function],
      }
    }
    getProducts={[MockFunction]}
    products={
      Object {
        "addMatchers": [Function],
        "advanceTimersByTime": [Function],
        "autoMockOff": [Function],
        "autoMockOn": [Function],
        "clearAllMocks": [Function],
        "clearAllTimers": [Function],
        "deepUnmock": [Function],
        "disableAutomock": [Function],
        "doMock": [Function],
        "dontMock": [Function],
        "enableAutomock": [Function],
        "fn": [Function],
        "genMockFromModule": [Function],
        "isMockFunction": [Function],
        "mock": [Function],
        "products": Array [
          Object {
            "id": 1,
            "name": "iphone",
          },
        ],
        "requireActual": [Function],
        "requireMock": [Function],
        "resetAllMocks": [Function],
        "resetModuleRegistry": [Function],
        "resetModules": [Function],
        "restoreAllMocks": [Function],
        "retryTimes": [Function],
        "runAllImmediates": [Function],
        "runAllTicks": [Function],
        "runAllTimers": [Function],
        "runOnlyPendingTimers": [Function],
        "runTimersToTime": [Function],
        "setMock": [Function],
        "setTimeout": [Function],
        "spyOn": [Function],
        "unmock": [Function],
        "useFakeTimers": [Function],
        "useRealTimers": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
